src/app/clusters/ota-requestor/OTARequestor.cpp

```diff
@@ -107,12 +107,40 @@ void OTARequestor::InitState(intptr_t context)
     OTARequestor * requestorCore = reinterpret_cast<OTARequestor *>(context);
     VerifyOrDie(requestorCore != nullptr);
 
-    // This results in the initial periodic timer kicking off
-    requestorCore->RecordNewUpdateState(OTAUpdateStateEnum::kIdle, OTAChangeReasonEnum::kSuccess);
+    if (requestorCore->mCurrentUpdateState != OTAUpdateStateEnum::kApplying)
+    {
+        // This results in the initial periodic timer kicking off
+        requestorCore->RecordNewUpdateState(OTAUpdateStateEnum::kIdle, OTAChangeReasonEnum::kSuccess);
+    }
+    else
+    {
+        // This may have been a reboot from applying an image
```

Comment:
It would be good to have a comment here that would described how the periodic timer will be kicked off in this logic path

Reply:
Done